src/main/java/net/datafaker/idnumbers/SwedenIdNumber.java

```diff
@@ -91,19 +95,48 @@ public static boolean isValidSwedishSsn(String ssn) {
         return checksum == calculatedChecksum;
     }
 
-    private static boolean parseDate(String ssn) {
+    private static boolean isDateIncorrect(String ssn) {
         String dateString = ssn.substring(0, 6);
-        if (ChronoField.YEAR.range().isValidIntValue(Integer.parseInt(dateString.substring(0, 2)))) {
-            if (ChronoField.MONTH_OF_YEAR.range().isValidIntValue(Integer.parseInt(dateString.substring(2, 4)))) {
-                if (ChronoField.DAY_OF_MONTH.range().isValidIntValue(Integer.parseInt(dateString.substring(4)))) {
-                    LocalDate date = LocalDate.parse(dateString, DATE_TIME_FORMATTER);
-                    // want to check that the parsed date is equal to the supplied data, most of the attempts will fail
-                    String reversed = date.format(DATE_TIME_FORMATTER);
-                    return !reversed.equals(dateString);
-                }
+
+        if (!ChronoField.YEAR.range().isValidIntValue(Integer.parseInt(dateString.substring(0, 2)))) {
+            return true;
+        }
+
+        if (!ChronoField.MONTH_OF_YEAR.range().isValidIntValue(Integer.parseInt(dateString.substring(2, 4)))) {
+            return true;
+        }
+
+        if (!ChronoField.DAY_OF_MONTH.range().isValidIntValue(Integer.parseInt(dateString.substring(4)))) {
+            return true;
+        }
+
+        char symbol = ssn.charAt(6);
+
+        if (symbol == '+') {
+            dateString = "19" + dateString;
```

Comment:
Yeah, you're right that's way better!

but, in fact this is a problem of swedish SSN itself. The format is not ready for 22 century so it will eventually break in the future (or rather ssn format will got change in the future).